### PR TITLE
feat: add arithmetic function "power" with decimal type

### PR DESCRIPTION
**Pull request by anshuldata**
Thank you for submitting a PR!

Before you continue, please ensure that your PR title and description (this message!) follow [conventional commit syntax](1). Substrait uses an automated release process that, among other things, uses PR titles & descriptions to build a changelog, so the syntax and format matter!

The title of the PR should be a valid commit header.

Some examples of proper commit message headers and PR titles:

 - `feat: add feature X`
 - `fix: X in case of Y`
 - `docs: improve documentation for X`

Note the case and grammar conventions.

Furthermore, the description of any PR that includes a breaking change should contain a paragraph that starts with `BREAKING CHANGE: ...`, where `...` explains what changed. The automated release process uses this to determine how it should bump the version number. Anything that changes the behavior of a plan that was previously legal is considered a breaking change; note that this includes behavior specifications that only exist in Substrait in the form of behavior descriptions on the website or in comments.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: extensions/functions_arithmetic_decimal.yaml
Comment (EpsilonPrime) [inline extensions/functions_arithmetic_decimal.yaml:261]: integer -> decimal (and elsewhere)

File: extensions/functions_arithmetic_decimal.yaml
Comment (EpsilonPrime) [inline extensions/functions_arithmetic_decimal.yaml:177]: no need for a blank line between function definitions

File: extensions/functions_arithmetic_decimal.yaml
Comment (EpsilonPrime) [inline extensions/functions_arithmetic_decimal.yaml:219]: Will the precision and scale stay the same as a result of this operation or do we need to extend it?  I suppose overflow is quickly likely so it's not a big deal if we don't rescale.

File: extensions/functions_arithmetic_decimal.yaml
Comment (EpsilonPrime) [inline extensions/functions_arithmetic_decimal.yaml:179]: This the aggregate functions section.  These definitions should be placed in the scalar functions section.

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: fixed it, Thanks

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: fixed it

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: fixed it

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: max it will be Decimal(37, S), which  supports factorial of max 33 (verified in duckdb and Snowflake). In both these engines type of return datatype was max size and didn't depend on input size. So I made it Decimal(38, S)

**Review by EpsilonPrime — DISMISSED** [review]

**Comment by anshuldata** [comment]
@EpsilonPrime I don't have write access to branch. Can you kindly merge ?
Thanks

**Comment by EpsilonPrime** [comment]
As this PR is somewhere between a new feature and a change to an existing one I'd like to get another approval for this PR before we merge it.  I'll check with the other folks at the Substrait community meeting on Wednesday and will likely merge it then.

File: extensions/functions_arithmetic_decimal.yaml
Comment (jacques-n) [inline extensions/functions_arithmetic_decimal.yaml:119]: I don't think these data types are right. This says that both arguments must have the same precision and scale. I think this and all the ones below are not right.

File: extensions/functions_arithmetic_decimal.yaml
Comment (jacques-n) [inline extensions/functions_arithmetic_decimal.yaml:172]: Isn't this some pattern of max of others?

**Review by jacques-n — CHANGES_REQUESTED** [review]
I don't think the precision/scale handling is correct here.

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: Fixed power function. I have removed all other functions, I will add them in separate PR

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: I have add this function in separate PR. I will take care of this comment in that future PR

**Review by jacques-n — APPROVED** [review]
This lgtm. @EpsilonPrime ?

**Comment by jacques-n** [comment]
@anshuldata can you rebase?

**Comment by anshuldata** [comment]
> @anshuldata can you rebase?

rebased it

File: extensions/functions_arithmetic_decimal.yaml
Comment (anshuldata) [reply extensions/functions_arithmetic_decimal.yaml]: resolving since added this functions to #675

**Review by jacques-n — APPROVED** [review]

**Pull request merged**
